frontend/public/components/console-app.jsx

```diff
@@ -0,0 +1,116 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import * as UIActions from '../actions/ui';
+import CloudShell from '@console/app/src/components/cloud-shell/CloudShell';
+import store from '../redux';
+import AppContents from './app-contents';
+import useEventListener from './utils/useEventListener';
+import { Helmet } from 'react-helmet';
+import { Page } from '@patternfly/react-core';
+import { ConsoleNotifier } from './console-notifier';
+import { ConnectedNotificationDrawer } from './notification-drawer';
+import { getBrandingDetails, Masthead } from './masthead';
+import { Navigation } from './nav';
+
+export const ConsoleApp = ({ location, match }) => {
+  const breakpointMD = 768;
+  const NOTIFICATION_DRAWER_BREAKPOINT = 1800;
+
+  const isDesktop = () => window.innerWidth >= breakpointMD;
+  const isLargeLayout = () => window.innerWidth >= NOTIFICATION_DRAWER_BREAKPOINT;
+
+  const [previousNavOpen, setPreviousNavOpen] = React.useState(isDesktop());
+  const [isNavOpen, setIsNavOpen] = React.useState(isDesktop());
+
+  const [previousDrawerInline, setPreviousDrawerInline] = React.useState(isLargeLayout());
+  const [isDrawerInline, setIsDrawerInline] = React.useState(isLargeLayout());
+
+  const usePrevious = (value) => {
```

Comment:
Just to be clear, this is a React hook that has the following behavior:
- creates and maintains a reference to `value`
- returns the same object (`value`) on every render

Is that correct?

If so, I think we can simplify its code to:

```ts
const ref = React.useRef(value);
return ref.current;
```

or maybe even use `React.useRef` directly?

Reply:
Also, note that if you don't pass `[]` as dependencies array to `React.useEffect` then the code will run on *every* component render, not sure if we need that here.

Reply:
The point is to have previous render value, not the same object every render.

@glekner 
We have `usePrevious` hook in `console-shared`. Can you use that ?

Reply:
yep, fixed